source/models/index.py

```diff
@@ -0,0 +1,45 @@
+import os
+# from models.table import Table
+from outputs.exceptions import *
+
+class Index:
+    def __init__(self, table, index_value):
+        self.table = table
+        self.index_value = index_value
+
+    def serialize(self):
+        os.makedirs(self.get_path(), exist_ok = True)
+    
+    def get_path(self):
+        return os.path.join(self.table.get_path(), os.path.join("indices" , self.index_value))
+    
+    def add_primary_key(self, primary_key, index_value):
+        index_value_file_path = os.path.join(self.get_path(), index_value + ".txt")
+        old_primary_keys = set()
+        if os.path.isfile(index_value_file_path):
+            old_primary_keys = set(self.get_primary_key(index_value))                
+        old_primary_keys.add(primary_key)
+        Index.__write_in_file(path = index_value_file_path, data = old_primary_keys)
+    
+    def delete_primary_key(self, primary_key, index_value):
+        index_value_file_path = os.path.join(self.get_path(), index_value + ".txt")
+        if not os.path.isfile(index_value_file_path):
+            raise FileNotFound(message = "there is no file with this index : " + index_value)
+        old_primary_keys = self.get_primary_key(index_value)
+        old_primary_keys.remove(primary_key)
+        Index.__write_in_file(path = index_value_file_path, data = old_primary_keys)
+
+    def get_primary_key(self, index_value):
+        index_value_file_path = os.path.join(self.get_path(), index_value + ".txt")
```

Comment:
You should initialize primary keys as an input array list

Reply:
I don't understand what is the problem here and in the next 2 comments.
i save the PKs in txt file separated by "," and when I call ```get_PK``` it open the txt file and load the inside data into a set